hrms/overrides/employee_advance.py

```diff
@@ -0,0 +1,230 @@
+from hrms.hr.doctype.employee_advance.employee_advance import EmployeeAdvance
+from suite42_erp_app.overrides.custom_task import CustomTask, TaskType
+from erpnext.accounts.doctype.payment_entry.payment_entry import (
+    get_party_details,
+    get_account_details,
+)
+from suite42_erp_app.suite42_erp_app.doctype.suite42_application_config.suite42_application_config import (
+    Suite42ApplicationConfig,
+)
+from suite42_erp_app.suite42_erp_app.common.utils import user_has_role
+from erpnext.accounts.doctype.sales_invoice.sales_invoice import get_bank_cash_account
+from suite42_erp_app.suite42_erp_app.common.authentication.role import (
+    RoleEnum,
+    EmployeeAdvanceConstants,
+)
+from suite42_erp_app.suite42_erp_app.common.decorators import (
+    handle_exceptions_with_readable_message,
+)
+from frappe.utils import cstr, flt
+import frappe
+from frappe import _
+
+
+class CustomEmployeeAdvance(EmployeeAdvance):
+    def validate(self):
+        self.validate_approver()
+        old_doc = self.get_doc_before_save()
+
+        if old_doc is not None and old_doc.status == self.status:
+            if self.status == "Draft":
+                if frappe.session.user != self.owner:
+                    frappe.throw(_("Only Owner can edit in Draft State"))
+            elif self.status == "Pending Approval":
+                if not user_has_role(frappe.session.user, RoleEnum.EXPENSE_APPROVER1_ROLE.value):
```

Comment:
check if approver_1 user == frappe.session.user instead of role

Reply:
Done